packages/flutter_tools/lib/src/codegen.dart

```diff
@@ -148,8 +148,7 @@ class CodeGeneratingKernelCompiler implements KernelCompiler {
       fileSystemScheme: kMultiRootScheme,
       depFilePath: depFilePath,
       targetModel: targetModel,
-      // Pass an invalid file name to prevent frontend_server from initializing from dill.
-      initializeFromDill: 'none_file',
+      initializeFromDill: initializeFromDill,
```

Comment:
How come we now want the dill whereas previously we specifically didnt?

Reply:
It took me a while to track down this bug - switching between no-builders/builders changes the dill contents in compatible ways due to the multiroot scheme. Adding this logic to the fingerprinter makes the invalidation automatic.